src/bank01_player.asm

```diff
@@ -723,189 +771,141 @@ scrollRoom:
-    ld   A, $ff                                        ;; 01:455f $3e $ff
-    ld   [wNextRoomOverride], A                        ;; 01:4561 $ea $44 $c3
-    ld   [wNextRoomOverride.x], A                      ;; 01:4564 $ea $45 $c3
-    ld   A, $00                                        ;; 01:4567 $3e $00
-.jr_01_4569:
-    and  A, $0f                                        ;; 01:4569 $e6 $0f
-    jr   NZ, .jr_01_4587                               ;; 01:456b $20 $1a
-    ld   A, [wScrollPixelCounter]                      ;; 01:456d $fa $48 $c3
-    swap A                                             ;; 01:4570 $cb $37
-    and  A, $0f                                        ;; 01:4572 $e6 $0f
-    sub  A, $0a                                        ;; 01:4574 $d6 $0a
-    cpl                                                ;; 01:4576 $2f
-    ld   E, A                                          ;; 01:4577 $5f
-    ld   D, $ff                                        ;; 01:4578 $16 $ff
-    call drawRoomMetaTilesColumn                       ;; 01:457a $cd $90 $46
-    ld   A, [wBackgroundDrawPositionX]                 ;; 01:457d $fa $42 $c3
-    dec  A                                             ;; 01:4580 $3d
-    dec  A                                             ;; 01:4581 $3d
-    and  A, $1f                                        ;; 01:4582 $e6 $1f
-    ld   [wBackgroundDrawPositionX], A                 ;; 01:4584 $ea $42 $c3
-.jr_01_4587:
-    ld   A, $b1                                        ;; 01:4587 $3e $b1
-    call scrollMoveSprites_trampoline                  ;; 01:4589 $cd $29 $04
-    pop  DE                                            ;; 01:458c $d1
-    ld   A, D                                          ;; 01:458d $7a
-    cpl                                                ;; 01:458e $2f
-    inc  A                                             ;; 01:458f $3c
-    ld   E, A                                          ;; 01:4590 $5f
-    ld   D, $00                                        ;; 01:4591 $16 $00
-    call scrollRoomScroll                              ;; 01:4593 $cd $c4 $46
-    ret                                                ;; 01:4596 $c9
+.scrollRoomMove:
+    ld   HL, wScrollPixelCounter
+    inc  C ; $00 for first call, $01 for subsequent call
+    swap C ; $00 for first call, $10 for subsequent call
+    xor  A, A
+    bit  DIRECTIONB_WEST, E
+    jr   NZ, .east
+    bit  DIRECTIONB_NORTH, E
+    jr   NZ, .south
+    bit  DIRECTIONB_SOUTH, E
+    jr   NZ, .north
+.west: ; by default if the others fell through
+    ld   E, D
+    ld   D, A
+    ld   A, E
+    cpl
+    inc  A
+    ld   E, A
+    ld   A, $0f
+    and  A, [HL]
+    jr   NZ, .west_done_graphics
+    ; Queue up the next section of metatiles
+    ld   A, [HL]
+    add  A, C ; adds $10 to queue the next section instead of the section actively being scrolled into
+    swap A
+    and  A, $0f
+    sub  A, $0a
```

Comment:
I would replace this constant with `(SCRN_X_B / 2)`.

It would be better to have constants for screen width and height, but I think for this PR that replacement will do fine.

Reply:
Done